packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
```

Comment:
not sure if this is confusing or not, but a dataset's id is different from its item id. there's a lot of place in the code base where we split the dataset id to get the item id.

Reply:
Good point - we should leave `content.id` to mean `item.id` and a few parameters

- `content.hubId` our own id, e.g. `content.hubId = '${item.id}_${item.values.layerIndex}'`
- `content.slug` e.g. `urban-forestry-trees`

Reply:
seems like we should also add `content.layerId` (or `content.layerIndex`)?

Reply:
@tomwayson the `layerId` is very content type specific. It's nonsensical for documents/apps and confusing for maps!. 

Related _specifically_ to Feature Service layers there is a change to Online in 8.2 (July) that expands services into layers. We should [review the docs](https://confluencewikidev.esri.com/display/AGO/Sub-Layer+Pages) to understand how it might work in our API and also indexers.

Reply:
OK, for `IFeatureLayerContent extents IContent`, seems like we should also add `featureLayerContent.layerId`